src/parse_instruction.py

```diff
@@ -295,15 +296,14 @@ def expect(n: str) -> str:
             value = AsmLiteral(parse_number(arg_elems))
         elif tok == "(":
             # Address mode or binary operation.
-            # There was possibly an offset, so value could be a AsmLiteral or Macro.
-            assert value is None or isinstance(value, (AsmLiteral, Macro))
```

Comment:
Was there a specific case you wanted to widen the type of `AsmAddressMode.lhs` for?

Reply:
Yes, stores to `N($sp)` and `(N + 4)($sp)` where N is an AsmGlobalSymbol at parse time. Not used yet though.